llvm/lib/Target/X86/X86ISelLoweringCall.cpp

```diff
@@ -669,8 +669,14 @@ const MCPhysReg *X86TargetLowering::getScratchRegisters(CallingConv::ID) const {
   return ScratchRegs;
 }
 
-ArrayRef<MCPhysReg> X86TargetLowering::getRoundingControlRegisters() const {
+ArrayRef<MCPhysReg>
+X86TargetLowering::getRoundingControlRegisters(const char *AsmStr) const {
   static const MCPhysReg RCRegs[] = {X86::FPCW, X86::MXCSR};
+  if (AsmStr) {
```

Comment:
I think any attempt at parsing the content of the string is a bad path to go down. Why is this worthwhile over what was here before, and just unconditionally add it?

Reply:
This is from the performance consideration. People uses inline asm for performance sometimes.

Reply:
I would say those people are wrong, but if you're using strictfp you've thrown that out. I would drop this. If you insist, it needs to be a separate review

Reply:
No strong objections, removed.

Reply:
But I'd argue that strictfp has two independent controls: "rounding mode" and "exceptions". It's a reasonable usage to only care about "rounding mode" and ignore "exceptions". For this case, scheduling non-RC-def inline asm among other FP instructions is permitted.